File: euclidean_algorithm/EuclideanAlgorithm.rb
require 'time'

class EuclideanAlgorithm
  a = 221
  b = 143

  start_time = Time.now

  loop do
    r = a % b

    break if r == 0

    a = b
    b = r
  end

  puts "Greatest common factor is '#{b}'"

  end_time = Time.now

  puts "Exec time: #{(end_time - start_time) * 100000} ms"
end

File: euclidean_algorithm/EuclideanAlgorithm.java
import java.util.*;

/**
 * EuclideanAlgorithm
 */
public class EuclideanAlgorithm {

  public static void main(String[] args) {
    int a = 221;
    int b = 143;

    long start = System.nanoTime();

    while(true) {
      int r = a % b;

      if(r == 0) {
        System.out.println("Greatest common factor is '" + b + "'");
        break;
      }

      a = b;
      b = r;
    }

    long end = System.nanoTime();

    System.out.println("Exec time: " + (end - start) / 1000000f + " ms");
  }
}